catchup/ledgerFetcher.go

```diff
@@ -166,20 +166,9 @@ func (lf *ledgerFetcher) getPeerLedger(ctx context.Context, peer network.HTTPPee
 			return fmt.Errorf("getPeerLedger received a tar header with data size of %d", header.Size)
 		}
 		balancesBlockBytes := make([]byte, header.Size)
-		readComplete := int64(0)
-
-		for readComplete < header.Size {
-			bytesRead, err := tarReader.Read(balancesBlockBytes[readComplete:])
-			readComplete += int64(bytesRead)
-			if err != nil {
-				if err == io.EOF {
-					if readComplete == header.Size {
-						break
-					}
-					err = fmt.Errorf("getPeerLedger received io.EOF while reading from tar file stream prior of reaching chunk size %d / %d", readComplete, header.Size)
-				}
-				return err
-			}
+		_, err = io.ReadFull(tarReader, balancesBlockBytes)
```

Comment:
This works fine, just noting we do not have an _automated_ test that covers the positive case of `getPeerLedger`. Fully understand this gets exercised manually in the meantime.

Reply:
There are tests in catchup/service_test.go that exercise the successful cases of getPeerLedger (and the catchup service generally) downloading blocks. You can see the coverage report here: https://app.codecov.io/gh/algorand/go-algorand/compare/4003/tree/catchup